boards/arm/mps2_an521/board.cmake

```
@@ -21,5 +25,5 @@ if (CONFIG_BUILD_WITH_TFM)
 elseif (CONFIG_SOC_MPS2_AN521_CPU1)
   set(CPU0_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/zephyr/boards/arm/mps2_an521/empty_cpu0-prefix/src/empty_cpu0-build/zephyr)
   set(QEMU_KERNEL_OPTION "-device;loader,file=${CPU0_BINARY_DIR}/zephyr.elf")
-  set(QEMU_EXTRA_FLAGS "-device;loader,file=${CMAKE_CURRENT_BINARY_DIR}/zephyr/zephyr.elf")
+  list(APPEND QEMU_EXTRA_FLAGS -device;loader,file=${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME})
```

minor nit, you don't need the `;` when the values to the `APPEND` is not quoted.

Secondly, removing the quotes can give unexpected behavior if any of the `PROJECT_BINARY_DIR` or `KERNEL_ELF_NAME` contains a space.

Therefore one of the following proposals are safe in this case:
```suggestion
  list(APPEND QEMU_EXTRA_FLAGS -device "loader,file=${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME}")
```
or
```suggestion
  list(APPEND QEMU_EXTRA_FLAGS "-device;loader,file=${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME}")
```

---

Fixed!

---

Hi @tejlmand ,

I was wondering if you may revisit this topic this week.